modules/sdk-coin-sol/test/unit/sol.ts

```diff
@@ -1482,5 +1497,80 @@ describe('SOL:', function () {
       should.equal(durableNonceTxnJson.feePayer, testData.SolInputData.pubKey);
       should.equal(durableNonceTxnJson.numSignatures, testData.SolInputData.durableNonceSignatures);
     });
+
+    it('should recover a txn for unsigned sweep recoveries', async function () {
+      // Unsigned Sweep Recovery
+      const unsignedSweepTxn = await basecoin.recover({
+        bitgoKey: testData.keys.bitgoKey,
+        recoveryDestination: testData.keys.destinationPubKey,
+        walletPassphrase: testData.keys.walletPassword,
+        durableNoncePK: testData.keys.durableNoncePubKey,
+        durableNonceSK: testData.keys.durableNoncePrivKey,
+      });
+
+      unsignedSweepTxn.should.not.be.empty();
+      unsignedSweepTxn.should.hasOwnProperty('serializedTx');
+
+      const unsignedSweepTxnDeserialize = new Transaction(coin);
+      unsignedSweepTxnDeserialize.fromRawTransaction(unsignedSweepTxn.serializedTx);
+      const unsignedSweepTxnJson = unsignedSweepTxnDeserialize.toJson();
+
+      should.equal(unsignedSweepTxnJson.nonce, testData.SolInputData.durableNonceBlockhash);
+      should.equal(unsignedSweepTxnJson.feePayer, testData.SolInputData.pubKey);
+      should.equal(unsignedSweepTxnJson.numSignatures, testData.SolInputData.unsignedSweepSignatures);
+    });
+
+    it('should handle error in recover function if a required field is missing/incorrect', async function () {
+      // missing userkey
+      await basecoin
```

Comment:
We could potentially break these up into separate unit tests, but it looks good for now!